TP1/Correction/count_words.py

```diff
@@ -0,0 +1,28 @@
+"""Count the number of words in a file."""
+
+from collections import Counter
+
+import utils
+
+DEFAULT_WORD_TO_SEARCH = "python"
+
+
+def get_word_occurrence(text: str, word: str, case_sensitive: bool = False) -> int:
+    """Return the number of times `word` appears in `text`."""
+    if not case_sensitive:
+        text = text.lower()
+        word = word.lower()
+    words_counter = Counter(text.split())
```

Comment:
Ca reste un tricks à connaitre pour moi d'utiliser un Counter, on pourrait donner une variante:

```python
def find_number_of_occurrences(text: str, word: str) -> int:
    start_index = 0
    number_of_occurrences = 0
    while start_index != -1:
        start_index = text.find(word, start_index)
        number_of_occurrences += 1
    return number_of_occurrences

Reply:
Ouais ok effectivement je vais leur mettre ca, et mettre `Counter` en commentaire

Reply:
I went for something even simpler